### DISCUSSION TITLE
"Error updating component prop"

### DISCUSSION THREAD

**Discussion by JohannesM007** [Budibase/budibase#11212] [Help] [answered]
Hi folks,
Unfortunately, I am no longer able to update some of the screens I have created. 
No matter what I want to change, it always appears the error message: 

"Error updating component prop"
![image](https://github.com/Budibase/budibase/assets/125383022/c6a778df-a141-49af-88ae-be037b621d50)

Latest version: 2.8.10. also 2.8.2 was not working ( self hosted )
Browser: Google Chrome / Mozilla Firefox
Also if i create a new app without any rows it fails!

What is the reason for this?
![image](https://github.com/Budibase/budibase/assets/125383022/4bbb0e47-8046-4b77-9fe8-710ac9982758)

I am able to change / edit some of my screens but some not!?
Regards Johannes
![image](https://github.com/Budibase/budibase/assets/125383022/61351ad2-aae5-452b-abbe-8c4a5adfe604)

**Comment by melohagan** [comment]
Hey @JohannesM007 

It is possible that your app database has become conflicted somehow. Are you able to follow the steps outlined here? https://github.com/Budibase/budibase/discussions/7777#discussioncomment-3732390

If not feel free to email a download link of your app export to support@budibase.com and I can take a look if you wish.

**Reply by JohannesM007** [reply]
Hi @melohagan,
thank you for your offer to check the app.
I just sent the link with the app to support@budibase.com.
Regards, Johannes

**Reply by ConorWebb96** [reply, marked as answer]
Hey @JohannesM007,

Firstly thank you for providing the app export. This was very helpful. I've gone through and had a look at the screens you highlighted in the screenshot above, below is a list of how I fixed them.

The general answer to all the issues you are experiencing is that you are nesting components that shouldn't be nested inside one another. With the last update validation that was previous not working was fixed meaning that going forward illegal nesting such as the below can no longer be done.

- Products page - you have a form component inside another form component.
- Complaints page - you have a side panel inside another side panel.
- Complaints/:id - is form nested inside another form.
- /complaints/new/row - form nested inside another form.
- samples - lots of forms nested in forms, I think theres 3-4 on this page.
- samples/:id - a form nested within another form.

**Reply by JohannesM007** [reply]
Thank you for your feedback, so i guess i have to redesign all this screens. Bad news but i will try to fix it.
Regards, Johannes

**Comment by dannyshin76** [comment]
@melohagan I am having the same problem now after upgrading my self-host to from v2.1.24 to v2.8.10, even after I cleaned up all conflicts on DB table as explained on #7777.
With some experimentation, I think I've successfully narrowed down the cause, which is "nested forms".
Nested forms were allowed in earlier versions obviously, but it seems it's causing problems in recent version's designer screen with imported apps created in earlier versions. Is this situation intended or just a bug? (Because the imported old apps with nested forms run as expected without problem on v2.8.10, I suspect it is just a bug on the designer screen.)

**Reply by ConorWebb96** [reply]
Hey @dannyshin76,

You are right going forward there are certain components that can no longer be nested inside one another. Forms is one of these components. Another would be side panels inside another side panel. 

I hope this makes sense!

**Reply by dannyshin76** [reply]
That is a bad news, because it means many users (including me) will have to completely rewrite many pages with nested forms. I don't know other people's use case but for me, I have to use nested forms to implement a mean to share or communicate information between forms. Also I can't mangle multiple forms into one because in my use case, one form must represent a single JSON object with schema. So I would like to suggest a designer option to disable that validation. Will it be possible? I am asking this because my nested forms are being executed without problem even with the most recent upgrade.

**Reply by ConorWebb96** [reply]
Hey @dannyshin76,

You should be able to nest forms that are contained within a side panel (Meaning you still have access to the previous forms data). If you are self-host you can revert back to before the most recent (v2.7) update were this validation wasn't turned on.

Additionally, you may be able to use the custom form schema option to map out all your data before it is submitted.

I hope this helps.

**Reply by dannyshin76** [reply]
OK, thanks for your information. For now I guess I have no choice but to revert BB version to v2.7.x. I see v2.7.38 is the last one before upping to v2.8.x. Is this version OK?
